File: MenuVoting.WebApi/MenuVotingApp/src/app/models/menu.ts
export class Menu {
  id: string | null;
  components: number[] | null;
  menuPoolId: string | null;
  constructor(id: string | null, components: number[] | null, menuPoolId: string | null) {
    this.id = id;
    this.components = components;
    this.menuPoolId = menuPoolId;
  }
}

File: MenuVoting.Domain/Entities/Menu.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace MenuVoting.Domain.Entities;
public class Menu : IKeyedEntity<Guid>
{
	public Guid Id { get; set; }
	public ICollection<string>? Dishes { get; set; }
	[ForeignKey(nameof(Entities.MenuPool))]
	public Guid MenuPoolId { get; set; }
	[JsonIgnore]
	public MenuPool? MenuPool { get; set; }
	virtual public ICollection<Vote>? Votes { get; set; }
}
